File: Practice/big_is_great.cpp
#include <iostream>
#include <string>

int main()
{
    std::string str;
    int tests;
    std::cin >> tests;
    while(tests--)
    {
        std::cin >> str;
        if(std::next_permutation(str.begin(), str.end()))
           std::cout << str << std::endl;
        else
            std:: cout << "no answer" << std::endl;
    }
    return 0;

}

File: codesprint/big_is_great.py
def main(tests):
    for dummy in range(tests):
        s = input()
        if invariant(s):
            print('no answer')
        else:    
            work_horse(s)

def invariant(x):

    i = 0
    while i < len(x) - 1:
        if x[i] < x[i + 1]:
            return False
        i += 1
    return True

def work_horse(string):

    leng = len(string)
    l = leng - 2
    while l >= 0:
        if string[l] < string[l + 1]:
            break
        l -= 1

    i = l
    prev = '{'
    at = 0
    while l < leng:
        if string[l] < prev and string[l] > string[i]:
            prev = string[l]
            at = l
        l += 1
    #print(string[i], string[at])

    lst = list(string)
    tmp = lst[i]
    lst[i] = lst[at]
    lst[at] = tmp

    for char in lst[ : i + 1]:
        print(char, end = '')
    for char in sorted(lst[i + 1 :]):
        print(char, end = '')
    print()


tsts = int(input())
main(tsts)

